staging/src/k8s.io/kubectl/pkg/cmd/cp/cp.go

```diff
@@ -352,9 +353,10 @@ func stripPathShortcuts(p string) string {
 }
 
 func makeTar(srcPath, destPath string, writer io.Writer) error {
-	// TODO: use compression here?
-	tarWriter := tar.NewWriter(writer)
+	gzWriter := gzip.NewWriter(writer)
+	tarWriter := tar.NewWriter(gzWriter)
 	defer tarWriter.Close()
+	defer gzWriter.Close()
```

Comment:
nit: each defer should immediately follow the matching `NewWriter` call (in case of panic in the next call)

Reply:
that's true, I tried to close writers in order tough. Will push the change

Reply:
good point: we can explicitly close them in order at the end of the method, with defer only as fallback?

I would probably focus on the test failure first though, anyhow.